corehq/apps/case_search/xpath_functions/subcase_functions.py

```diff
@@ -98,25 +120,12 @@ def _get_parent_case_ids_matching_subcase_query(subcase_query, context):
         .source(['indices.referenced_id', 'indices.identifier'])
     )
 
-    counts_by_parent_id = Counter(
-        index['referenced_id']
-        for subcase in es_query.run().hits
-        for index in subcase['indices']
-        if index['identifier'] == subcase_query.index_identifier
-    )
-    if len(counts_by_parent_id) > MAX_RELATED_CASES:
-        from ..exceptions import TooManyRelatedCasesError
-        raise TooManyRelatedCasesError(
-            _("The related case lookup you are trying to perform would return too many cases"),
-            serialize(subcase_query.subcase_filter)
-        )
-
-    if subcase_query.op == '>' and subcase_query.count <= 0:
-        return list(counts_by_parent_id)
-
-    return [
-        case_id for case_id, count in counts_by_parent_id.items() if subcase_query.filter_count(count)
-    ]
+    if context.profiler:
```

Comment:
This seems to be the only place you check if the profiler is set, should the other usages also check?

Reply:
Good point, I was wondering about this as well.  The other usages don't need to check, because profiler is always set, but it's not reliably present on `SearchFilterContext` (that's initialized in `build_filter_from_xpath`, which is used in places outside case search).  It does seem a bit tricky to intuit, so maybe I should init a profiler in `build_filter_from_xpath` if one isn't provided, so we can always count on it being present.

Reply:
Oh right, there was another reference to `context.profiler` that I forgot to guard.  Thanks tests.  I guess that answers my question...